mopidy_local/schema.py

```diff
@@ -236,7 +236,7 @@ def dates(c, format="%Y-%m-%d"):
             operator.itemgetter(0),
             c.execute(
                 """
-        SELECT DISTINCT strftime(?, date) AS date
+        SELECT DISTINCT(strftime(?, substr(date || '-01-01', 1, 10))) AS date
```

Comment:
I don't think this is going to work for many cases, such as when a non-year is used here. Do you think you can add some tests to make sure that old behaviour is maintained?

Reply:
Dates in the yyyy-mm-dd format will still work as they did. Jibberish will still be jibberish. This seems ok to me but tests would be nice. I was going to push some before merging.

Reply:
What values are you concerned about ? Let's add tests for those specifically.

Reply:
I did run tests with my local music collection and it worked for both date formats.
I guess I could add some tests as well.

Reply:
So concern I have is that the `date` column is being converted into a valid iso format timestamp by using a string manipulation, instead of using `date` functions.

I'm not that concerned about it working because I see that it takes advantage of months and days starting both by `01` and filling in the gaps, however maybe a couple of tests that make sure that under those two date formats this code works.

Reply:
A problem with this is that it breaks the `format` param by always assuming "YYYY-MM-DD". In reality we only ever use "YYYY-MM-DD" so maybe we should just remove that param for simplicity.

Reply:
Nope! Wrong again! The `format` param works just fine, it's only used to format the resulting date result, not on the input.

Reply:
The format parameter determines the date format which is **returned**.
You can use `format="%Y"` for example to retrieve only the year part.
However, the second parameter "time-value" must conform to these formats: https://www.sqlite.org/lang_datefunc.html#time_values which are ISO formats.